server/src/main/java/org/elasticsearch/index/engine/CombinedDeletionPolicy.java

```diff
@@ -296,10 +317,20 @@ private static Set<String> listOfNewFileNames(IndexCommit previous, IndexCommit
     }
 
     /**
-     * Checks whether the deletion policy is holding on to acquired index commits
+     * Checks whether the deletion policy is holding on to externally acquired index commits
      */
     synchronized boolean hasAcquiredIndexCommits() {
-        return acquiredIndexCommits.isEmpty() == false;
+        if (Assertions.ENABLED) {
```

Comment:
Given that this method is for testing only, we can simply do what is inside this if-statement all the time. Since if assertions are disabled, the `internallyAcquiredIndexCommits` is empty anyway.

Reply:
Yes, that's indeed seems superfluous